src/plugin.ts

```diff
@@ -437,6 +490,9 @@ function buildPrecompileOptions<EnvSpecificOptions>(
     },
   };
 
+  const locals: string[] = [];
+  output.plugins?.ast?.push(discoverLocals(target, state, locals, scope));
```

Comment:
we push our own locals discovery ast plugin, which mutates the locals passed to precompile.
@chancancode if you have a better option, now is the chance :)